public/app/core/components/RolePicker/RolePickerMenu.tsx

```diff
@@ -467,6 +414,110 @@ export const RoleMenuOption = React.forwardRef<HTMLDivElement, React.PropsWithCh
 
 RoleMenuOption.displayName = 'RoleMenuOption';
 
+interface RoleMenuGroupsSectionProps {
+  roles: Role[];
+  renderedName: string;
+  menuSectionStyle: string;
+  groupHeaderStyle: string;
+  optionBodyStyle: string;
+  showGroups?: boolean;
+  optionGroups: Array<{
+    name: string;
+    options: Role[];
+    value: string;
+  }>;
+  onChange: (value: string) => void;
+  onOpenSubMenuRMGS: (value: string) => void;
+  onCloseSubMenu?: (value: string) => void;
+  groupSelected: (group: string) => boolean;
+  groupPartiallySelected: (group: string) => boolean;
+  disabled?: boolean;
+  subMenuNode?: HTMLDivElement;
+  showSubMenu: boolean;
+  openedMenuGroup: string;
+  subMenuOptions: Role[];
+  selectedOptions: Role[];
+  onChangeSubMenu: (option: Role) => void;
+  onClearSubMenu: () => void;
+  showOnLeftSubMenu: boolean;
+}
+
+export const RoleMenuGroupsSection = React.forwardRef<HTMLDivElement, RoleMenuGroupsSectionProps>(
+  (
+    {
+      roles,
+      renderedName,
+      menuSectionStyle,
+      groupHeaderStyle,
+      optionBodyStyle,
+      showGroups,
+      optionGroups,
+      onChange,
+      groupSelected,
+      groupPartiallySelected,
+      onOpenSubMenuRMGS,
+      onCloseSubMenu,
+      subMenuNode,
+      showSubMenu,
+      openedMenuGroup,
+      subMenuOptions,
+      selectedOptions,
+      onChangeSubMenu,
+      onClearSubMenu,
+      showOnLeftSubMenu,
+    },
+    _ref
+  ) => {
+    return (
+      <div>
+        {roles.length > 0 && (
+          <div className={menuSectionStyle}>
+            <div className={groupHeaderStyle}>{renderedName}</div>
+            <div className={optionBodyStyle}></div>
+            {showGroups && !!optionGroups?.length
+              ? optionGroups.map((option, i) => (
+                  <RoleMenuGroupOption
+                    data={option}
+                    key={i}
```

Comment:
```suggestion
                    key={option.uid}
```
Using array index as key should be avoided, it'd be something more stable like an id.

Reply:
Changed.